components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -182,18 +183,41 @@ void JsonRpcService::AddEthereumChain(mojom::EthereumChainPtr chain,
 void JsonRpcService::AddEthereumChainForOrigin(
     mojom::EthereumChainPtr chain,
     const GURL& origin,
-    AddEthereumChainCallback callback) {
+    AddEthereumChainForOriginCallback callback) {
   DCHECK_EQ(origin, url::Origin::Create(origin).GetURL());
   if (!origin.is_valid() ||
       add_chain_pending_requests_.contains(chain->chain_id) ||
       HasRequestFromOrigin(origin)) {
-    std::move(callback).Run(chain->chain_id, false);
+    std::move(callback).Run(chain->chain_id, false, std::string());
+    return;
+  }
+  RequestInternal(
+      eth::eth_chainId(), true, GURL(chain->rpc_urls.front()),
+      base::BindOnce(&JsonRpcService::OnEthChainIdValidated,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(chain), origin,
+                     std::move(callback)));
+}
+
+void JsonRpcService::OnEthChainIdValidated(
+    mojom::EthereumChainPtr chain,
+    const GURL& origin,
+    AddEthereumChainForOriginCallback callback,
+    const int http_code,
+    const std::string& response,
+    const base::flat_map<std::string, std::string>& headers) {
+  std::string result;
+  ParseSingleStringResult(response, &result);
```

Comment:
Could we check for an error here please. If there is an error we should call `ParseErrorResult<mojom::ProviderError>(` like the other cases to see if we can provide a better error.  See the other similar functions.